### ISSUE TITLE
Failed to scale volume 412 Precondition Failed RestrictedReplicaCount

### ISSUE THREAD

**Issue by todeb** [openebs/openebs#3795]
## Description
I want to scale volume replica to 1. Pod connected to volume is runnning on node6. So i expect that replica on node1 will be removed. But it is not.
Initially this volume had 1 replica on node1, I scal it to 2 replicas and then to 3 replicas and was able to scale it down to 2 replicas, but I cannot scale it to 1 replica.

```
kubectl mayastor get volume-replica-topology -n openebs 33c3ed9b-3684-47ad-988c-0801dcf23a62
 ID                                    NODE             POOL                 STATUS  CAPACITY  ALLOCATED  SNAPSHOTS  CHILD-STATUS  REASON  REBUILD
 ce2b1194-0d2d-4dd4-b8f9-0c7b27bf3b23  node6  pool-on-node6  Online  30GiB     13.3GiB    0 B        Online        <none>  <none>
 0a9cab43-2b09-47b9-8496-1b04fc8bdaab  node1  pool-on-node1  Online  30GiB     13.3GiB    0 B        Online        <none>  <none>


kubectl mayastor get volumes -n openebs
 ID                                    REPLICAS  TARGET-NODE      ACCESSIBILITY  STATUS  SIZE   THIN-PROVISIONED  ALLOCATED  SNAPSHOTS  SOURCE
 33c3ed9b-3684-47ad-988c-0801dcf23a62  2         node6  nvmf           Online  30GiB  true              13.3GiB    0          <none>

kubectl mayastor -n openebs scale volume 33c3ed9b-3684-47ad-988c-0801dcf23a62 1
Failed to scale volume 33c3ed9b-3684-47ad-988c-0801dcf23a62. Error error in response: status code '412 Precondition Failed', content: 'RestJsonError { details: "", message: "SvcError :: RestrictedReplicaCount: Replica Count of 1 is not attainable for AffinityGroup", kind: FailedPrecondition }'
```

## Expected Behavior
replica on node1 removed

## Current Behavior
error

## Your Environment
openebs v2.7.0

**Comment by tiagolobocastro** [comment]
> RestrictedReplicaCount: Replica Count of 1 is not attainable for AffinityGroup

CC @Abhinandan-Purkait

**Comment by Abhinandan-Purkait** [comment]
Yeah, so this is an expected behaviour at the moment, for the stsaffinity feature.

We don't have a mechanism to ensure stsAffinity feature's consistency when you scale up and down to 1, so we restrict the operation. We have control on the scheduling while creating and not removal. So, if someone goes up in replicas when stsAffinity feature is in enabled, and then tries to come down to 1 again, it is not possible at the moment to selective remove those extra replicas while ensuring the feature still holds.

**Comment by todeb** [comment]
> We don't have a mechanism to ensure stsAffinity feature's consistency when you scale up and down to 1
can't be just use the same function as it is when creating volumes with stsaffinity enabled?

If it is allowed to scale it up and there is a mechanism for this it should be allowed to scale it down with same mechanism as well.
If there is no mechanism, scale up probably should also not be allowed?

**Comment by tiagolobocastro** [comment]
@Abhinandan-Purkait can you remind me again how the rules work here for multi-replica volumes with `stsAffinityGroup` ?

For 1-replica volumes, with anti-affinity among the replica nodes, ensuring the replicas get placed on different nodes.
But IIRC for n-replica volumes, this changes somewhat and becomes more of a soft-rule or preference?

**Comment by Abhinandan-Purkait** [comment]
> [@Abhinandan-Purkait](https://github.com/Abhinandan-Purkait) can you remind me again how the rules work here for multi-replica volumes with `stsAffinityGroup` ?
> 
> For 1-replica volumes, with anti-affinity among the replica nodes, ensuring the replicas get placed on different nodes. But IIRC for n-replica volumes, this changes somewhat and becomes more of a soft-rule or preference?

For n-replicas the it's a soft rule but we try to spread it accross as many pools as possible, ex:- if 3Volume in a AffinityGroup and 3replicas each, and we have 9 pools, we will try to spread replicas accross 9 pools.

**Comment by tiagolobocastro** [comment]
I'll take a look to see if we can go down to 1 replica again if the restrictions are met
